Alignment/ReferenceTrajectories/plugins/CombinedTrajectoryFactory.cc

```diff
@@ -38,7 +38,15 @@ class CombinedTrajectoryFactory : public TrajectoryFactoryBase {
   CombinedTrajectoryFactory *clone() const override { return new CombinedTrajectoryFactory(*this); }
 
 private:
-  std::vector<TrajectoryFactoryBase *> theFactories;
+  CombinedTrajectoryFactory(const CombinedTrajectoryFactory &other)
+      : TrajectoryFactoryBase(other), theUseAllFactories{other.theUseAllFactories} {
+    theFactories.reserve(other.theFactories.size());
+    for (const auto &f : other.theFactories) {
+      theFactories.emplace_back(f->clone());
+    }
+  }
+
+  std::vector<std::unique_ptr<TrajectoryFactoryBase>> theFactories;
```

Comment:
This vector could be also done with `std::shared_ptr`, which would avoid defining the custom copy constructor (with the cloning), and would keep the `TrajectoryFactoryBase` objects shared between the copies of `CombinedTrajectoryFactory`.

Reply:
@makortel are you satisfied with this PR in its present status or are you planning  further update?

Reply:
I'm satisfied, just wanted to say that it is possible to also share the `TrajectoryFactoryBase` objects if that is needed. So I'm not planning updates.